examples/ner_new/main_train.py

```diff
@@ -0,0 +1,257 @@
+config_data = yaml.safe_load(open("configs/config_data.yml", "r"))
+config_model = yaml.safe_load(open("configs/config_model.yml", "r"))
+config_preprocess = \
+    yaml.safe_load(open("configs/config_preprocessor.yml", "r"))
+
+config = Config({}, default_hparams=None)
+config.add_hparam('config_data', config_data)
+config.add_hparam('config_model', config_model)
+config.add_hparam('preprocessor', config_preprocess)
+
+device = torch.device("cuda") if torch.cuda.is_available() \
+    else torch.device("cpu")
+
+tp_request = {
+    "scope": Sentence,
+    "schemes": {
+        "text_tag": {
+            "entry_type": Token,
+            "vocab_method": "indexing",
+            "attribute_get": "text",
+            "type": DATA_INPUT,
+            "extractor": AttributeExtractor
+        },
+        "char_tag": {
+            "entry_type": Token,
+            "vocab_method": "indexing",
+            "max_char_length": config.config_data.max_char_length,
+            "type": DATA_INPUT,
+            "extractor": CharExtractor
+        },
+        "ner_tag": {
+            "entry_type": EntityMention,
+            "attribute": "ner_type",
+            "based_on": Token,
+            "vocab_method": "indexing",
+            "type": DATA_OUTPUT,
+            "extractor": BioSeqTaggingExtractor
+        }
+    }
+}
+
+# All not specified dataset parameters are set by default in Texar.
+# Default settings can be found here:
+# https://texar-pytorch.readthedocs.io/en/latest/code/data.html#texar.torch.data.DatasetBase.default_hparams
+tp_config = {
+    "preprocess": {
+        "pack_dir": config.config_data.train_path,
+        "device": device.type
+    },
+    "dataset": {
+        "batch_size": config.config_data.batch_size_tokens
+    }
+}
+
+ner_train_reader = CoNLL03Reader(cache_in_memory=True)
+ner_val_reader = CoNLL03Reader(cache_in_memory=True)
+
+train_preprocessor = TrainPreprocessor(train_reader=ner_train_reader,
+                                       request=tp_request,
+                                       config=tp_config)
+
+model: nn.Module = \
+    create_model(schemes=train_preprocessor.feature_resource["schemes"],
+                 config=config)
+model.to(device)
+
+optim: Optimizer = SGD(model.parameters(),
+                       lr=config.config_model.learning_rate,
+                       momentum=config.config_model.momentum,
+                       nesterov=True)
+
+predictor = Predictor(batch_size=train_preprocessor.config.dataset.batch_size,
+                      model=model,
+                      predict_forward_fn=predict_forward_fn,
+                      feature_resource=train_preprocessor.feature_resource,
+                      cross_pack=False)
+evaluator = CoNLLNEREvaluator()
+
+val_pl = Pipeline()
+val_pl.set_reader(ner_val_reader)
+val_pl.add(predictor)
+val_pl.add(evaluator)
+
+epoch = 0
+train_err: float = 0.0
+train_total: float = 0.0
+train_sentence_len_sum: int = 0
+val_scores: List = []
```

Comment:
I think several of these variables are not used?

Reply:
`val_scores` is removed. All others are used.